devices.js

```diff
@@ -17260,16 +17260,46 @@ const devices = [
         model: 'HLU2909K',
         vendor: 'Datek',
         description: 'APEX smart plug 16A',
-        fromZigbee: [fz.electrical_measurement, fz.on_off, fz.metering],
+        fromZigbee: [fz.on_off, fz.electrical_measurement, fz.metering, fz.temperature],
         toZigbee: [tz.on_off],
         meta: {configureKey: 1},
         configure: async (device, coordinatorEndpoint, logger) => {
             const endpoint = device.getEndpoint(1);
-            await reporting.bind(endpoint, coordinatorEndpoint, ['genOnOff', 'seMetering']);
-            endpoint.saveClusterAttributeKeyValue('seMetering', {divisor: 100, multiplier: 1});
-            endpoint.saveClusterAttributeKeyValue('haElectricalMeasurement', {acVoltageDivisor: 100});
+            await reporting.bind(endpoint, coordinatorEndpoint, ['genOnOff', 'haElectricalMeasurement', 'msTemperatureMeasurement']);
+            await endpoint.read('haElectricalMeasurement', ['acVoltageMultiplier', 'acVoltageDivisor']);
+            await endpoint.read('haElectricalMeasurement', ['acCurrentMultiplier', 'acCurrentDivisor']);
+            await endpoint.read('haElectricalMeasurement', ['acPowerMultiplier', 'acPowerDivisor']);
+            await reporting.onOff(endpoint);
+            await reporting.rmsVoltage(endpoint);
+            await reporting.rmsCurrent(endpoint);
+            await reporting.activePower(endpoint);
+            await reporting.temperature(endpoint);
+        },
+        exposes: [e.power(), e.current(), e.voltage(), e.switch(), e.temperature()],
+    },
+
+    {
+        zigbeeModel: ['HAN'],
+        model: 'Datek HAN',
```

Comment:
Is there any model written on the device? (one where you can find the device with via Google?)

Reply:
Nop, I did ask the manufacturer but the model name are DATEK HAN. That is why I in the device information have provided URL to the product page.

Reply:
@Koenkk Can this be merged please (you said within hours) so I can get the OTA available to upgrade the HAN adapter, will be more available + fixing stability issues in this old fw.